tests/stdlib/talgorithm.nim

```diff
@@ -113,3 +113,153 @@ block:
     doAssert binarySearch(moreData, 6) == -1
     doAssert binarySearch(moreData, 4711) == 4
     doAssert binarySearch(moreData, 4712) == -1
+
+# merge
+block:
+  var x = @[1, 7, 8, 11, 21, 33, 45, 99]
+  var y = @[6, 7, 9, 12, 57, 66]
+
+  var merged: seq[int]
+  merged.merge(x, y)
+  doAssert merged.isSorted
+  doAssert merged == sorted(x & y)
+
+block:
+  var x = @[111, 88, 76, 56, 45, 31, 22, 19, 11, 3]
+  var y = @[99, 85, 83, 82, 69, 64, 48, 42, 33, 31, 26, 13]
+
+  var merged: seq[int]
+  merged.merge(x, y, proc (x, y: int): int = -system.cmp(x, y))
+  doAssert merged.isSorted(proc (x, y: int): int = -system.cmp(x, y))
+  doAssert merged == sorted(x & y, SortOrder.Descending)
+
+block:
+  var x: seq[int] = @[]
+  var y = @[1]
+
+  var merged: seq[int]
+  merged.merge(x, y)
+  doAssert merged.isSorted
+  doAssert merged.isSorted(SortOrder.Descending)
+  doAssert merged == @[1]
+
+block:
+  var x = [1, 3, 5, 5, 7]
+  var y: seq[int] = @[]
+
+  var merged: seq[int]
+  merged.merge(x, y)
+  doAssert merged.isSorted
+  doAssert merged == @x
+
```

Comment:
general rule: use templates to make tests more DRY (which makes them easier to evolve); eg:
```nim
template fn[T](x,y,e: seq[T]) =
  var merged: seq[T]
  merged.merge(x, y)
  doAssert merged.isSorted
  doAssert merged == e
fn([1, 3, 5, 5, 7], @[], [1, 3, 5, 5, 7])
fn(seq[int].default, @[1], @[1])
```

etc. this can be customized, eg adding `cmp2 = system.cmp[T]`

Reply:
Can be done in the future.